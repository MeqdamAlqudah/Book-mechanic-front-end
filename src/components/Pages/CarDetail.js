import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import AxiosWrapper from '../../requirments/AxiosWrapper';

const CarDetail = () => {
  const [currentCar, setCurrentCar] = useState({});
  const carId = useSelector((el) => el.carDetailReducer.currentCarId || JSON.parse(localStorage.getItem('carId')));
  useEffect(() => {
    localStorage.setItem('carId', JSON.stringify(carId));
    AxiosWrapper(`http://127.0.0.1:3000/api/v1/cars/${carId}`).then((res) => {
      setCurrentCar(res.data);
    });
  }, []);
  if (Object.keys(currentCar).length <= 0) {
    return (<div>loading...</div>);
  }
  return (
    <div className="main">
      <h1>Detail page</h1>
      Brand:
      {' '}
      {currentCar.brand}
      <br />
      {' '}
      Model:
      {' '}
      {currentCar.model}
      <br />
      Registration:
      {' '}
      {currentCar.registration}
      <br />
      Photo:
      {' '}
      <img src={currentCar.photo} alt="car" />

    </div>
  );
};

export default CarDetail;
