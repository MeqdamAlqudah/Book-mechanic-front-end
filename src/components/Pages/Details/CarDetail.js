import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import AxiosWrapper from '../../../requirments/AxiosWrapper';
import style from './style.module.css';

const CarDetail = ({ userId }) => {
  const [currentCar, setCurrentCar] = useState({});
  const carId = useSelector((el) => el.carDetailReducer.currentCarId || JSON.parse(localStorage.getItem('carId')));

  useEffect(() => {
    localStorage.setItem('carId', JSON.stringify(carId));
    AxiosWrapper(`http://127.0.0.1:3000/api/v1/users/${userId}/cars/${carId}`).then((res) => {
      setCurrentCar(res.data);
    });
  }, [carId]);
  if (Object.keys(currentCar).length <= 0) {
    return (<div>loading...</div>);
  }
  return (
    <div className={style.page}>
      <div className={style.container}>
        <section className={style['image-section']}>
          <h1 className={style.title}>Detail page</h1>
          Brand:
          {' '}
          {currentCar.brand}
          <br />
          {' '}
          Model:
          {' '}
          {currentCar.model}
          <br />
          Registration:
          {' '}
          {currentCar.registration}
          <br />
          Photo:
          {' '}
          <img src={currentCar.photo} className={style.image} alt="car" />
        </section>
      </div>

      <button type="button" className={style.btn}>
        <Link className={style.text} to="/appointment">Reserve</Link>
      </button>

    </div>
  );
};
CarDetail.propTypes = {
  userId: PropTypes.number,
};
CarDetail.defaultProps = {
  userId: 0,
};

export default CarDetail;
