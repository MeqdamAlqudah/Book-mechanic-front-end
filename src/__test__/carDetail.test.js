import * as React from 'react';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import CarDetail from '../components/Pages/CarDetail';
import store from '../redux/configureStore';

const renderer = require('react-test-renderer');

global.React = React;
describe('Test CarDetail', () => {
  test('test CarDetail rendering ', () => {
    const useSelector = jest.fn(() => 1);
    const AxiosWrapper = jest.fn(() => Promise.resolve({
      json: () => Promise.resolve({
        brand: 'BMW',
        model: 'Gt-200',
        photo: 'http/photo',
        registration: '232423-sad',
      }),
    }));

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <CarDetail userId={1} />
        </Router>
      </Provider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
