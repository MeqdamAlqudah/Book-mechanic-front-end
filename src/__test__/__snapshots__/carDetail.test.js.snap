// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CarDetail test CarDetail rendering  1`] = `
<div>
  loading...
</div>
`;
